include/gsl/span

```diff
@@ -451,62 +494,74 @@ public:
     constexpr span& operator=(const span& other) noexcept = default;
 
     // [span.sub], span subviews
-    template <std::ptrdiff_t Count>
+    template <std::size_t Count>
     constexpr span<element_type, Count> first() const noexcept
     {
-        Expects(Count >= 0 && Count <= size());
+        Expects(Count != dynamic_extent);
+        Expects(Count <= size());
```

Comment:
`Count <= size()` already tests that it isn't `dynamic_extent` because `dynamic_excent` is `SIZE_MAX`.

Reply:
(Occurs below)